src/App.jsx

```diff
@@ -14,45 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_PARAM_ALPHABET = 'alphabet';
+const SORT_PARAM_LENGTH = 'length';
+
+function getPreparedGood(goods, { sortParam, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortParam) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortParam) {
+        case SORT_PARAM_ALPHABET:
+          return goodA.localeCompare(goodB);
+        case SORT_PARAM_LENGTH:
+          return goodA.length - goodB.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortParam, setSortParam] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGood(
+    goodsFromServer,
+    {
+      sortParam,
+      isReversed,
+    },
+  );
+  const reset = () => {
+    setSortParam('');
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortParam(SORT_PARAM_ALPHABET)}
+          className={cn('button', 'is-info', {
+            'is-light': sortParam !== SORT_PARAM_ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortParam(SORT_PARAM_LENGTH)}
+          className={cn('button', 'is-success', {
+            'is-light': sortParam !== SORT_PARAM_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
FYI: it's better to write callback inside `setState` to update it based on previous: [useState](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state)